challengeutils/annotations.py

```diff
@@ -0,0 +1,269 @@
+    def etag(self, value):
+        if value is None:
+            raise ValueError("etag must not be None")
+        self._etag = str(value)
+
+# TODO: Remove once synapseclient==2.1.0
+def to_synapse_annotations(
+        annotations: Annotations
+    ) -> typing.Dict[str, typing.Any]:
+    """Transforms a simple flat dictionary to a Synapse-style Annotation
+    object.
+    https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/annotation/v2/Annotations.html
+    """
+
+    if is_synapse_annotations(annotations):
+        return annotations
+    synapse_annos = {}
+
+    if not isinstance(annotations, Annotations):
+        raise TypeError("annotations must be a synapseclient.Annotations "
+                        "object with 'id' and 'etag' attributes")
+
+    synapse_annos['id'] = annotations.id
+    synapse_annos['etag'] = annotations.etag
+
+    nested_annos = synapse_annos.setdefault('annotations', {})
+    for key, value in annotations.items():
+        elements = to_list(value)
+        element_cls = _annotation_value_list_element_type(elements)
+        if issubclass(element_cls, str):
+            nested_annos[key] = {'type': 'STRING',
+                                 'value': elements}
+        elif issubclass(element_cls, bool):
+            nested_annos[key] = {'type': 'STRING',
+                                 'value': [str(e).lower() for e in elements]}
+        elif issubclass(element_cls, int):
+            nested_annos[key] = {'type': 'LONG',
+                                 'value': [str(e) for e in elements]}
+        elif issubclass(element_cls, float):
+            nested_annos[key] = {'type': 'DOUBLE',
+                                 'value': [str(e) for e in elements]}
+        elif issubclass(element_cls, (datetime.date, datetime.datetime)):
+            nested_annos[key] = {'type': 'TIMESTAMP_MS',
+                                 'value': [str(to_unix_epoch_time(e))
+                                           for e in elements]}
+        else:
+            nested_annos[key] = {'type': 'STRING',
+                                 'value': [str(e) for e in elements]}
+    return synapse_annos
+
+# TODO: Remove once synapseclient==2.1.0
+def from_synapse_annotations(
+        raw_annotations: typing.Dict[str, typing.Any]
+    ) -> Annotations:
+    """Transforms a Synapse-style Annotation object to a simple flat
+    dictionary."""
+    if not is_synapse_annotations(raw_annotations):
+        raise ValueError('Unexpected format of annotations from Synapse. '
+                         'Must include keys: "id", "etag", and "annotations"')
+
+    annos = Annotations(raw_annotations['id'], raw_annotations['etag'])
+    for key, value_and_type in raw_annotations['annotations'].items():
+        key: str
+        conversion_func = ANNO_TYPE_TO_FUNC[value_and_type['type']]
+        annos[key] = [conversion_func(v) for v in value_and_type['value']]
+
+    return annos
+
+
+def _convert_to_annotation_cls(
```

Comment:
Just wanted to double-check:
This function will be removed once synapseclient 2.1.0 is released as well?  Since everything it's dependent on will be removed?

Reply:
This function will actually remain unless I add it to the `synapseclient`